pkg/resource/deploy/deploytest/provider_test.go

```diff
@@ -0,0 +1,434 @@
+// Copyright 2016-2023, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package deploytest
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/blang/semver"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource/plugin"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/tokens"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestProvider(t *testing.T) {
+	t.Parallel()
+	t.Run("SignalCancellation", func(t *testing.T) {
+		t.Parallel()
+		t.Run("has CancelF", func(t *testing.T) {
+			t.Parallel()
+			var called bool
+			prov := &Provider{
+				CancelF: func() error {
+					called = true
+					return fmt.Errorf("expected error")
```

Comment:
Nit: this should be `errors.New`

Reply:
I thought I turned a linter on for this!?

Reply:
This is what perfsprint is supposed to pick up? Is this file not being linted at all?